From: Wanpeng Li <liwanp@linux.vnet.ibm.com>
Date: Tue, 17 Dec 2013 10:45:52 +1100
Subject: [PATCH] sched/numa: fix setting of cpupid on page migration twice

References: Automatic NUMA Balancing (fate#315482)
Patch-mainline: v3.14
Git-commit: baae911b27b8dbee6830f4e3ef0fcf4dc8e9c07b

Commit 7851a45cd3f6 ("mm: numa: Copy cpupid on page migration") copiess
over the cpupid at page migration time.  It is unnecessary to set it again
in migrate_misplaced_transhuge_page().

Signed-off-by: Wanpeng Li <liwanp@linux.vnet.ibm.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 mm/migrate.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/mm/migrate.c b/mm/migrate.c
index 48248de..85b628f 100644
--- a/mm/migrate.c
+++ b/mm/migrate.c
@@ -1758,8 +1758,6 @@ int migrate_misplaced_transhuge_page(struct mm_struct *mm,
 	if (!new_page)
 		goto out_fail;
 
-	page_cpupid_xchg_last(new_page, page_cpupid_last(page));
-
 	isolated = numamigrate_isolate_page(pgdat, page);
 	if (!isolated) {
 		put_page(new_page);
